src/nsrt_learning.py

```diff
@@ -309,3 +329,40 @@ def unify_effects_and_options(
             f_new_ground_delete_effects,
         f_lifted_option_args | f_new_lifted_add_effects | \
             f_new_lifted_delete_effects)
+
+
+@dataclass(repr=False)
+class _NSRTIntermediateData:
```

Comment:
hmm I really don't know how to think about this class. I think the fact that we have to call it NID is indicative of it not being a semantically coherent thing :P it's somewhere between a Partition and a partially-formed NSRT, and it's confusing for me to keep the relationship between all of these things straight.

I see the motivation though. can we talk about alternatives? some ideas:

1. make NSRTs not always frozen, but add a `freeze` method on them, which we call at the end of learning, and after which we can treat them as frozen. then build up NSRTs piece by piece, starting with just the effects, then adding the preconditions, then the option specs, etc. We could potentially get rid of `STRIPSOperator` if we do this....
2. extend Partition so that it basically becomes what NID is now.
3. others...